qmp_helper/src/qmp_helper.c

```diff
@@ -386,7 +393,13 @@ int main(int argc, char *argv[])
                 qmph_exit_cleanup(ret);
             }
             QMPH_LOG("Accepted the connection fd: %d, telling qemu.", qhs.unix_fd);
-            qmp_connect(&qhs);
+            ret = qmp_connect(&qhs);
+            if (ret == -1 && errno == ECONNREFUSED) {
```

Comment:
@eric-ch Do you think this should be only `if (ret == -1)`?  There is a failure in any case, so it didn't connect.  Closing the unix side makes sense.

Reply:
It could be `EINTR`?
Now that you point it out, except for `EINTR` (which should be handled around `argo_send` really), I cannot think of a case where it might be useful to keep the unix socket open is `argo_send()` failed.